Assets/HoloToolkit/UX/Scripts/BoundingBoxes/BoundingBoxGizmoShell.cs

```diff
@@ -0,0 +1,70 @@
+﻿//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+//
+
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace HoloToolkit.Unity.UX
+{
+    /// <summary>
+    /// Draws a bounding box gizmo in the style of the hololens shell
+    /// </summary>
+    [ExecuteInEditMode]
+    public class BoundingBoxGizmoShell : BoundingBoxGizmo
+    {
+        #region public
+
+        /// <summary>
+        /// If true, bounding box edges will display when active
+        /// To more closely mimic shell behavior, set this to false
+        /// </summary>
+        public bool DisplayEdgesWhenSelected = true;
+
+        /// <summary>
+        /// Whether to clamp handle size between HandleScaleMin / Max
+        /// This may result in bunched-up handles so use with caution
+        /// </summary>
+        public bool ClampHandleScale = false;
+
+        /// <summary>
+        /// User-defined min size for handle
+        /// Useful when using the bounding box with extremely small objects
+        /// This value is ignored if ClampHandleSize is false
+        /// </summary>
+        public float HandleScaleMin = 0.1f;
+
+        /// <summary>
+        /// User-defined min size for handle
+        /// Useful when using the bounding box with extremely large objects
+        /// This value is ignored if ClampHandleSize is false
+        /// </summary>
+        public float HandleScaleMax = 2.0f;
+
+        #endregion
+
+        #region protected
+
+        [SerializeField]
+        protected Renderer edgeRenderer;
+
+        [SerializeField]
+        protected GameObject xyzHandlesObject;
+        [SerializeField]
+        protected GameObject xyHandlesObject;
+        [SerializeField]
+        protected GameObject xzHandlesObject;
+        [SerializeField]
+        protected GameObject zyHandlesObject;
+
+        #endregion
+
+        #region custom editor
+#if UNITY_EDITOR
+        [UnityEditor.CustomEditor(typeof(BoundingBoxGizmoShell))]
+        public class CustomEditor : MRTKEditor  { }
```

Comment:
If this is empty, does it need to be here?

Reply:
Custom editors need to be defined for each type of class that they edit. It's a Unity thing.

Reply:
MRTKEditor  is not a Unity thing.

I'd really like us not to use this custom editor if we don't need to.

Reply:
No need to apologize, but I think this would be a good opportunity to open an issue where we can go over the MRTK editor in a more formal / centralized way. It's not very useful to outsiders if all the issues you have are raised piecemeal in a bunch of unrelated PRs.

Reply:
I've opened an issue here: #1589 

@NeerajW will merge this PR

Reply:
Aside from the redesign of the editor stuff itself, why does this empty class need to be declared if it's empty and unused?

Reply:
@StephenHodgson I completely agree with you that this is odd. In the spirit of helping PRs move along, we filed an issue where we can figure out what is MRTK editor (already merged in the past) and why is it needed. Unfortunately not sure why this was not caught sooner. For example: CompoundButtonMesh.cs also has this goop.

Reply:
I've updated the issue with a bit of documentation that explains the basics - let's pick up this discussion over there.